webhook/resourcesemantics/interface.go

```diff
@@ -28,3 +29,26 @@ type GenericCRD interface {
 	apis.Validatable
 	runtime.Object
 }
+
+// GenericCRDWithConfig is the interface definition that allows us to perform
+// the generic CRD actions like deciding whether to increment generation and so
+// forth. You can further customiz with SupportedVerbs and SupportedSubResources
+// which
+type GenericCRDWithConfig struct {
+	GenericCRD
+
+	// supportedVerbs are the verbs registered for the callback.
+	// If left empty, configures, Create, Update, and Delete
+	SupportedVerbs []admissionregistrationv1.OperationType
+
+	// supportedSubResources are the subresources that will be registered
+	// for the resource validation.
+	// To get the main resource and status registered (old behaviour), you
+	// leave this empty.
+	// If you wanted to add for example scale validation for Deployments, you'd
+	// do:
+	// []string{"", "/status", "/scale"}
+	// And to get just the main resource, you would do:
+	// []string{""}
+	SupportedSubResources []string
```

Comment:
An alternative to this, which requires wrapping our typical objects in this struct, might be to define these as accessor methods in a separate interface and use type assertions?

e.g.
```
type VerbLimited interface {
    SupportedVerbs() []admissionregistrationv1.OperationType
}

type SubResourceLimited iterface {
   SupportedSubResources() []string
}
```

Then you can have the logic say:
```go
supportedVerbs := []string{/* what we do today*/}
if vl, ok := blah.(apis.VerbLimited); ok {
  supportedVerbs = vl.SupportedVerbs()
}
```

Reply:
> The one downside is that then reusing esp. duck types will get a bit more complicated since the code defines what verbs/subresources are configured and changing them will require code changes.

Can you elaborate on this?

Reply:
Well, maybe I'm just confused about how this looks from the callers point of view. Aka, the person creating the webhook config in their main.

Say, there's a predefined `Resource`, like here (say it would be PodSpecable):
```
	handlers = map[schema.GroupVersionKind]resourcesemantics.GenericCRD{
		{
			Group:   "pkg.knative.dev",
			Version: "v1alpha1",
			Kind:    "Resource",
		}: &Resource{},
}
```

And then I want to say, nah, I actually don't want "status" updates for it. `Resource` is defined somewhere else, so I can't just add these methods to it in some other place (like my code that's using pkg types). So, I guess I'd have to define something like:
type ResourceCustom struct {
   GenericCRD
}

func (r *ResourceCustom) SupportedSubResources() []string {
  return []string{""} // just the main resource
}

And now the handlers call would be like so:
```
	handlers = map[schema.GroupVersionKind]resourcesemantics.GenericCRD{
		{
			Group:   "pkg.knative.dev",
			Version: "v1alpha1",
			Kind:    "Resource",
		}: &ResourceCustom{GenericCRD: &Resource{}},
}

@mattmoor @dprotaso is that what y'all are thinking, or am I missing something obvs?